libr/cons/cons.c

```diff
@@ -697,6 +697,19 @@ R_API int r_cons_get_cursor(int *rows) {
 	return col;
 }
 
+R_API bool r_cons_isatty() {
+#if __UNIX__ || __CYGWIN__
+	struct winsize win = { 0 };
+        if (isatty (1) && !ioctl (0, TIOCGWINSZ, &win)) {
+                if ((win.ws_col == 0) || (win.ws_row == 0)) {
```

Comment:
Right. And im checking stdin and stdout. When i should only check stdout. will update the pr later but thats the overall idea

> On 02 Jun 2016, at 20:15, Joshua J. Drake <notifications@github.com> wrote:
> 
> In libr/cons/cons.c:
> 
> > @@ -697,6 +697,19 @@ R_API int r_cons_get_cursor(int *rows) {
> >  	return col;
> >  }
> >  
> > +R_API bool r_cons_isatty() {
> > +#if __UNIX__ || __CYGWIN__
> > +	struct winsize win = { 0 };
> > +        if (isatty (1) && !ioctl (0, TIOCGWINSZ, &win)) {
> > +                if ((win.ws_col == 0) || (win.ws_row == 0)) {
> The logic appears to be broken here. Should it be !=0 && !=0 ??
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>